cmd/kubeadm/app/cmd/phases/reset/cleanupnode.go

```diff
@@ -100,17 +102,32 @@ func absoluteKubeletRunDirectory() (string, error) {
 		klog.Warningf("[reset] Failed to evaluate the %q directory. Skipping its unmount and cleanup: %v\n", kubeadmconstants.KubeletRunDirectory, err)
 		return "", err
 	}
-
-	// Only unmount mount points which start with "/var/lib/kubelet" or absolute path of symbolic link, and avoid using empty absoluteKubeletRunDirectory
-	umountDirsCmd := fmt.Sprintf("awk '$2 ~ path {print $2}' path=%s/ /proc/mounts | xargs -r umount", absoluteKubeletRunDirectory)
-	klog.V(1).Infof("[reset] Executing command %q", umountDirsCmd)
-	umountOutputBytes, err := exec.Command("sh", "-c", umountDirsCmd).Output()
+	err = unmountKubeletDirectory()
 	if err != nil {
-		klog.Warningf("[reset] Failed to unmount mounted directories in %s: %s\n", kubeadmconstants.KubeletRunDirectory, string(umountOutputBytes))
+		klog.Warningf("[reset] Failed to unmount mounted directories in %s \n", kubeadmconstants.KubeletRunDirectory)
+		return "", err
 	}
 	return absoluteKubeletRunDirectory, nil
 }
 
+func unmountKubeletDirectory() error {
```

Comment:
i think in the original ticket i mentioned:
> which only happens in a _linux.go file and the _windows.go file is a NOP.

so that we don't call this code on windows or osx (that apparently doesn't use /proc/mount too).

Reply:
+ i would add a description on top:
`// unmountKubeletDirectory unmounts all paths that contain KubeletRunDirectory`